src/util/mpir_hwtopo.c

```diff
@@ -230,7 +239,8 @@ int MPII_hwtopo_finalize(void)
     int mpi_errno = MPI_SUCCESS;
 
 #ifdef HAVE_HWLOC
-    hwloc_topology_destroy(hwloc_topology);
+    if (!from_pmi)
+        hwloc_topology_destroy(hwloc_topology);
```

Comment:
Who destroys the topology in the case of `PMIx_load_topology`? If it is destroyed in `PMIx_Finalize`, then it should be destroyed in `MPIR_pmi_finalize`. Here we call `MPIR_pmi_load_hwloc_topology`, that's it.

Reply:
I think topology from PMIx_load_topology does not need to be destroyed explicitly,  because it is returned from PMIx library. It is not correct to call hwloc_topology_destroy to it.

Reply:
It is destroyed in `PMIx_Finalize`. The PMI client library owns the topology.

Reply:
That's why this if statement. hwloc_topology_destroy is not called if hwloc_topology is obtained from PMI library.